### PR TITLE
feat(mobile): use local transactions for graph

### PR DESCRIPTION
**Pull request by Nodonisko**
## Description

This PR moves transaction-cache functionallity to redux, because in the end it's simplest solution. There two next steps that should be done, we should add function for calculating balance changes of ETH + tokens and optionally we may look into optimilizing performance of this solution (it works nice for normal accounts for for accounts with lot of transactions it could be slow).

## Related Issue

Resolve

### CODE REVIEW

File: suite-common/graph/src/graphBalanceEvents.ts [hunk 1]
```diff
@@ -107,18 +110,26 @@ export const getAccountMovementEvents = async ({
...
     startOfTimeFrameDate: Date | null;
     endOfTimeFrameDate: Date;
+    dispatch: ReturnType<typeof useDispatch>;
 }) => {
     const { coin, identity, descriptor } = account;
 
     const getBalanceHistory = async () => {
         if (getNetworkType(coin) === 'ripple') {
-            return TransactionCacheEngine.getAccountBalanceHistory({
+            const allTransactions = await dispatch(
```
Comment (matejkriz) [inline suite-common/graph/src/graphBalanceEvents.ts:124]: Why is this applied for ripple only?

File: suite-common/graph/src/balanceHistoryUtils.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            summary.txs += 1;
+            summary.received = amount.plus(receivedDrops);
+            summary.sent = amount.plus(sentDrops);
+        } else {
+            summaryMap.set(blockTime, {
+                time: blockTime,
+                txs: 1,
+                received: receivedDrops,
+                sent: sentDrops,
+                sentToSelf: new BigNumber(0),
```
Comment (matejkriz) [inline suite-common/graph/src/balanceHistoryUtils.ts]: With ripple is it not possible to send to self?

File: suite-common/graph/src/balanceHistoryUtils.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            sentDrops = amount;
+        } else if (tx.type === 'recv') {
+            receivedDrops = amount;
+        }
+
+        if (summaryMap.has(blockTime)) {
+            const summary = summaryMap.get(blockTime)!;
+            summary.txs += 1;
+            summary.received = amount.plus(receivedDrops);
+            summary.sent = amount.plus(sentDrops);
```
Comment (matejkriz) [inline suite-common/graph/src/balanceHistoryUtils.ts]: I don't get this 🤔
 if for example `tx.type === 'sent'`:
 ```
sentDrops = amount
receivedDrops = 0
// =>
summary.sent = amount + amount // ? 
```

File: suite-common/graph/src/balanceHistoryUtils.ts (hunk 2 above)
Comment (Nodonisko) [reply suite-common/graph/src/balanceHistoryUtils.ts]: Yes, from what I know ripple could have only one output.

File: suite-common/graph/src/balanceHistoryUtils.ts (hunk 3 above)
Comment (Nodonisko) [reply suite-common/graph/src/balanceHistoryUtils.ts]: Yea, this is definitely mistake.

**Comment by matejkriz** [comment]
already merged in another PR

**Pull request closed without merging**